awx/main/access.py

```diff
@@ -2196,6 +2206,7 @@ class NotificationTemplateAccess(BaseAccess):
 
     def filtered_queryset(self):
         return self.model.objects.filter(
+            Q(organization__in=Organization.objects.filter(notification_admin_role__members=self.user)) |
             Q(organization__in=self.user.admin_of_organizations) |
```

Comment:
I would prefer to consolidate the 2 first OR conditions by using organization `accessible_objects` based on `notification_admin_role`, which will be a superset of `admin_of_organizations`

Reply:
Note that this is the same logical path that you took when modifying the `can_read` method